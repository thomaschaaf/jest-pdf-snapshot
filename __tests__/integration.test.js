/**
 * @group integration
 */

/* eslint-disable global-require */
const shell = require('shelljs');
const fs = require('fs');

shell.config.silent = true;

function maskTime(jestOutput) {
  return jestOutput
    .replace(/\(\d+ ms\)/g, '(??? ms)')
    .replace(/\d+\.\d+ s, estimated \d+ s/g, '??? s')
    .replace(/\d+\.\d+ s/g, '??? s');
}

describe('jest-pdf-snapshot', () => {
  beforeAll(() => {
    shell.cd('__tests__/mock_project');
    shell.rm('-rf', 'node_modules');
    shell.exec('yarn install');
    shell.cd('..');
    shell.rm('-rf', 'sandbox');
  });

  beforeEach(() => {
    shell.cp('-r', './mock_project', './sandbox');
    shell.cd('./sandbox');
  });

  afterEach(() => {
    shell.cd('..');
    shell.rm('-rf', './sandbox');
  });

  it('works as expected when local testing', () => {
    // When
    const result = shell.exec('yarn test -i mock.test.js');

    // Then
    expect(maskTime(result.stderr)).toMatchSnapshot();
    expect(fs.existsSync('__pdf_snapshots__/mock-test-js-snapshot-is-absent-1.pdf'))
      .toBeTruthy();
    expect(fs.existsSync('__pdf_snapshots__/__diff_output__/mock-test-js-snapshot-is-different-1-diff.pdf'))
      .toBeTruthy();
  });

  it('works as expected when updating snapshot', () => {
    // Given
    const snapshotPath = '__pdf_snapshots__/mock-test-js-snapshot-is-different-1.pdf';
    const originalModifiedTime = fs.statSync(snapshotPath).mtimeMs;

    // When
    const result = shell.exec('yarn test -i mock.test.js --update-snapshot');

    // Then
    expect(maskTime(result.stderr)).toMatchSnapshot();
    expect(fs.existsSync('__pdf_snapshots__/mock-test-js-snapshot-is-absent-1.pdf'))
      .toBeTruthy();

    const newModifiedTime = fs.statSync(snapshotPath).mtimeMs;
    expect(newModifiedTime).toBeGreaterThan(originalModifiedTime);
  });

  it('works as expected in ci', () => {
    // When
    const result = shell.exec('yarn test -i mock.test.js --ci');

    // Then
    expect(maskTime(result.stderr)).toMatchSnapshot();
  });

  it('works as expected for error cases', () => {
    // When
    const result = shell.exec('yarn test -i error.test.js');

    // Then
    expect(maskTime(result.stderr)).toMatchSnapshot();
  });
});
