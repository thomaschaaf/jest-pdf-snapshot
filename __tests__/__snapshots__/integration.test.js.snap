// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-pdf-snapshot works as expected for error cases 1`] = `
"[0m[7m[1m[32m PASS [39m[22m[27m[0m [2m./[22m[1merror.test.js[22m
  [32m✓[39m [2mNot pdf buffer (??? ms)[22m

[1m[32m › 1 snapshot written.[39m[22m
[1mSnapshot Summary[22m
[1m[32m › 1 snapshot written [39m[22mfrom 1 test suite.

[1mTest Suites: [22m[1m[32m1 passed[39m[22m, 1 total
[1mTests:       [22m[1m[32m1 passed[39m[22m, 1 total
[1mSnapshots:   [22m[1m[32m1 written[39m[22m, 1 total
[1mTime:[22m        ??? s
[2mRan all test suites[22m[2m matching [22m/error.test.js/i[2m.[22m
"
`;

exports[`jest-pdf-snapshot works as expected in ci 1`] = `
"[0m[7m[1m[31m FAIL [39m[22m[27m[0m [2m./[22m[1mmock.test.js[22m
  [32m✓[39m [2mSnapshot matches (??? ms)[22m
  [31m✕[39m [2mSnapshot is absent (??? ms)[22m
  [31m✕[39m [2mSnapshot is different (??? ms)[22m

[1m[31m  [1m● [22m[1mSnapshot is absent[39m[22m

    New snapshot was [1m[31mnot written[39m[22m. The update flag must be explicitly passed to write a new snapshot.

     + This is likely because this test is run in a continuous integration (CI) environment in which snapshots are not written by default.
[2m[22m
[2m    [0m [90m 12 | [39m[0m[22m
[2m    [0m [90m 13 | [39mtest([32m'Snapshot is absent'[39m[33m,[39m () [33m=>[39m {[0m[22m
[2m    [0m[31m[1m>[22m[2m[39m[90m 14 | [39m  expect(pdfBuffer1)[33m.[39mtoMatchPdfSnapshot()[33m;[39m[0m[22m
[2m    [0m [90m    | [39m                     [31m[1m^[22m[2m[39m[0m[22m
[2m    [0m [90m 15 | [39m})[33m;[39m[0m[22m
[2m    [0m [90m 16 | [39m[0m[22m
[2m    [0m [90m 17 | [39mtest([32m'Snapshot is different'[39m[33m,[39m () [33m=>[39m {[0m[22m
[2m[22m
[2m      [2mat Object.test ([22m[2m[0m[36mmock.test.js[39m[0m[2m:14:22)[22m[2m[22m

[1m[31m  [1m● [22m[1mSnapshot is different[39m[22m

    Expected pdf to be same as the snapshot, but was different.
    [1m[31mSee diff for details:[39m[22m [31m/home/zeptometer/code/Tsingtao/__tests__/sandbox/__pdf_snapshots__/__diff_output__/mock-test-js-snapshot-is-different-1-diff.pdf[39m
[2m[22m
[2m    [0m [90m 16 | [39m[0m[22m
[2m    [0m [90m 17 | [39mtest([32m'Snapshot is different'[39m[33m,[39m () [33m=>[39m {[0m[22m
[2m    [0m[31m[1m>[22m[2m[39m[90m 18 | [39m  expect(pdfBuffer2)[33m.[39mtoMatchPdfSnapshot()[33m;[39m[0m[22m
[2m    [0m [90m    | [39m                     [31m[1m^[22m[2m[39m[0m[22m
[2m    [0m [90m 19 | [39m})[33m;[39m[0m[22m
[2m    [0m [90m 20 | [39m[0m[22m
[2m[22m
[2m      [2mat Object.test ([22m[2m[0m[36mmock.test.js[39m[0m[2m:18:22)[22m[2m[22m

[1m[31m › 2 snapshots failed.[39m[22m
[1mSnapshot Summary[22m
[1m[31m › 2 snapshots failed[39m[22m from 1 test suite. [2mInspect your code changes or run \`yarn test -u\` to update them.[22m

[1mTest Suites: [22m[1m[31m1 failed[39m[22m, 1 total
[1mTests:       [22m[1m[31m2 failed[39m[22m, [1m[32m1 passed[39m[22m, 3 total
[1mSnapshots:   [22m[1m[31m2 failed[39m[22m, [1m[32m1 passed[39m[22m, 3 total
[1mTime:[22m        [1m[33m??? s[39m[22m
[2mRan all test suites[22m[2m matching [22m/mock.test.js/i[2m.[22m
[2K[1G[31merror[39m Command failed with exit code 1.
"
`;

exports[`jest-pdf-snapshot works as expected when local testing 1`] = `
"[0m[7m[1m[31m FAIL [39m[22m[27m[0m [2m./[22m[1mmock.test.js[22m
  [32m✓[39m [2mSnapshot matches (??? ms)[22m
  [32m✓[39m [2mSnapshot is absent (??? ms)[22m
  [31m✕[39m [2mSnapshot is different (??? ms)[22m

[1m[31m  [1m● [22m[1mSnapshot is different[39m[22m

    Expected pdf to be same as the snapshot, but was different.
    [1m[31mSee diff for details:[39m[22m [31m/home/zeptometer/code/Tsingtao/__tests__/sandbox/__pdf_snapshots__/__diff_output__/mock-test-js-snapshot-is-different-1-diff.pdf[39m
[2m[22m
[2m    [0m [90m 16 | [39m[0m[22m
[2m    [0m [90m 17 | [39mtest([32m'Snapshot is different'[39m[33m,[39m () [33m=>[39m {[0m[22m
[2m    [0m[31m[1m>[22m[2m[39m[90m 18 | [39m  expect(pdfBuffer2)[33m.[39mtoMatchPdfSnapshot()[33m;[39m[0m[22m
[2m    [0m [90m    | [39m                     [31m[1m^[22m[2m[39m[0m[22m
[2m    [0m [90m 19 | [39m})[33m;[39m[0m[22m
[2m    [0m [90m 20 | [39m[0m[22m
[2m[22m
[2m      [2mat Object.test ([22m[2m[0m[36mmock.test.js[39m[0m[2m:18:22)[22m[2m[22m

[1m[32m › 1 snapshot written.[39m[22m
[1m[31m › 1 snapshot failed.[39m[22m
[1mSnapshot Summary[22m
[1m[32m › 1 snapshot written [39m[22mfrom 1 test suite.
[1m[31m › 1 snapshot failed[39m[22m from 1 test suite. [2mInspect your code changes or run \`yarn test -u\` to update them.[22m

[1mTest Suites: [22m[1m[31m1 failed[39m[22m, 1 total
[1mTests:       [22m[1m[31m1 failed[39m[22m, [1m[32m2 passed[39m[22m, 3 total
[1mSnapshots:   [22m[1m[31m1 failed[39m[22m, [1m[32m1 written[39m[22m, [1m[32m1 passed[39m[22m, 3 total
[1mTime:[22m        ??? s
[2mRan all test suites[22m[2m matching [22m/mock.test.js/i[2m.[22m
[2K[1G[31merror[39m Command failed with exit code 1.
"
`;

exports[`jest-pdf-snapshot works as expected when updating snapshot 1`] = `
"[0m[7m[1m[32m PASS [39m[22m[27m[0m [2m./[22m[1mmock.test.js[22m
  [32m✓[39m [2mSnapshot matches (??? ms)[22m
  [32m✓[39m [2mSnapshot is absent (??? ms)[22m
  [32m✓[39m [2mSnapshot is different (??? ms)[22m

[1m[32m › 3 snapshots updated.[39m[22m
[1mSnapshot Summary[22m
[1m[32m › 3 snapshots updated [39m[22mfrom 1 test suite.

[1mTest Suites: [22m[1m[32m1 passed[39m[22m, 1 total
[1mTests:       [22m[1m[32m3 passed[39m[22m, 3 total
[1mSnapshots:   [22m[1m[32m3 updated[39m[22m, 3 total
[1mTime:[22m        ??? s
[2mRan all test suites[22m[2m matching [22m/mock.test.js/i[2m.[22m
"
`;
